### PR TITLE
feat(ControllerEvents): get controller magnitude for swing speed

### PR DESCRIPTION
**Pull request by pargee**
Apologies if this isn't done right -- my first pull request on GH! This was brought up as a question on Slack (and one of my own yesterday). It might be too self-explanatory, but as someone who didn't know Vector3.Magnitude was available, I'm hoping to remove any headaches others who want to use this for items like sword swing detection and other motion-based gestures.

Also removed a few double semicolons in the existing code.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: e421af72843e6bf449084a6e0080d12b8b69fbb0
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/thestonefox/SteamVR_Unity_Toolkit

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: e421af72843e6bf449084a6e0080d12b8b69fbb0
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 2eedea656bb37a614dd7c01facb0e435723f1b8b
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/thestonefox/SteamVR_Unity_Toolkit

---
This message was auto-generated by https://gitcop.com

**Comment by mattboy64** [comment]
Looks like GitCop got ya again ;)

File: Assets/VRTK/Scripts/VRTK_ControllerEvents.cs [hunk 1]
```diff
@@ -575,6 +576,15 @@ public Vector3 GetAngularVelocity()
         }
 
         /// <summary>
+        /// The GetMagnitude method is used to return the current magnitude of the physical game controller. This can be useful for determining the speed at which the controller is being swung.
+        /// </summary>
+        /// <returns>A float illustrating the length of the vector based on current real world physical controller velocity.</returns>
+        public float GetMagnitude() {
```
Comment (mattboy64) [inline Assets/VRTK/Scripts/VRTK_ControllerEvents.cs]: curly bracket needs to be on it's own line.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: e421af72843e6bf449084a6e0080d12b8b69fbb0
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 2eedea656bb37a614dd7c01facb0e435723f1b8b
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/thestonefox/SteamVR_Unity_Toolkit

---
This message was auto-generated by https://gitcop.com

File: Assets/VRTK/Scripts/VRTK_ControllerEvents.cs (hunk 1 above)
Comment (pargee) [inline Assets/VRTK/Scripts/VRTK_ControllerEvents.cs]: Thanks, good catch! I had it set but VS snuck it back to inline.

**Comment by thestonefox** [comment]
What's the benefit of having this method as an explicit thing in the script?

Can you not just do this yourself by getting the Velocity and then calling magnitude on the return in your own code?

**Comment by pargee** [comment]
It would just be a quality of life change - feel free to discard if necessary. I've always needed the magnitude rather than raw velocity for the interactions I've built so far (the same with using a Rigidbody's velocity.magnitude).

I went through and originally did the formula for getting magnitude from velocity sqrt(x² + y² + z²); not knowing there was a vec3 operation (since mag has always been fed to me :P), so I thought this might be helpful to others and save them 10 mins of poking how to calc mag from vel, heh.

It's simple enough as you said, so I can just keep it in my own  helper if you'd rather not expand with those types of features :) Thanks for lookin!

**Comment by thestonefox** [comment]
I'll ask in the slack channel, we'll get an idea of what people think about the feature, see if it's something people would want

**Comment by mattboy64** [comment]
Square root isn't the fastest thing in the world, so calculating this often could have some performance considerations (see https://docs.unity3d.com/ScriptReference/Vector3-sqrMagnitude.html).

If this stays, I'd pull it out of the SetVelocity() function....actually you can just do this and avoid the step completely:

```
public float GetMagnitude()
{
      return controllerVelocity.magnitude;
}

public float GetMagnitudeSqr()
{
    return controllerVelocity.sqrMagnitude;
}
``` 

Looking at it know, it seems a little unnecessary since it's only one line though......either way.

**Comment by thestonefox** [comment]
@mattboy64 is it worth having those additional methods though that literally just call the same method again?

**Comment by mattboy64** [comment]
Probably not since it's kind of redundant.

**Comment by thestonefox** [comment]
Is it worth just closing this PR then?

**Comment by mattboy64** [comment]
The only situation where I could see this being useful is if the user didn't know that the magnitude of the velocity is what they need. That's going to be useful for some people, but I don't how many, or if it's worth spelling out. What did slack say?

**Comment by thestonefox** [comment]
I didn't ask yet, i should I suppose.

**Comment by mattboy64** [comment]
My vote is to just close it out. I appreciate the contribution, but keeping the code base lean and simple feels more important to me, especially since this would just be a one line function.

**Pull request closed without merging**